From 6e0698737e68597f265fa18f75d2a6405a6228b5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 9 Oct 2017 17:59:30 +0100
Subject: [PATCH 3/3] core: Disable disastrous NET_WM_PING deletion dialogs

These cause more hassle than they're worth, and break many SDL games.
These apps are just slower to start (Thunderbird) or GL-locked (SDL),
and throwing up a "not responsive" window to break focus, input, etc,
when it clearly *is* working, just isn't helpful to anyone.

It should also be noted that many of these apps do not *voluntarily*
subscribe to NET_WM_PING, so to destroy them for not responding to an
advertised capability is a major regression.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>

diff --git a/src/core/delete.c b/src/core/delete.c
index 81537df..5c8807a 100644
--- a/src/core/delete.c
+++ b/src/core/delete.c
@@ -50,6 +50,8 @@ meta_window_ensure_close_dialog (MetaWindow *window)
 {
   MetaDisplay *display;
 
+  return;
+
   if (window->close_dialog)
     return;
 
@@ -64,6 +66,12 @@ void
 meta_window_show_close_dialog (MetaWindow *window)
 {
   meta_window_ensure_close_dialog (window);
+
+  if (!window->close_dialog)
+  {
+    return;
+  }
+  
   meta_close_dialog_show (window->close_dialog);
 
   if (window->display &&